protocols/ping/src/handler.rs

```diff
@@ -273,7 +273,7 @@ impl ConnectionHandler for Handler {
         loop {
             // Check for outbound ping failures.
             if let Some(error) = self.pending_errors.pop_back() {
-                log::debug!("Ping failure: {:?}", error);
+                tracing::debug!("Ping failure: {:?}", error);
```

Comment:
I believe tracing has a way of capturing errors, we should use that too!

Reply:
updated to `tracing::error!` so that this event gets captured at the error level

Reply:
Sorry for not being more clear here, that is not what I meant :)

Logging this is as an `tracing::error!` is way too verbose, we don't want to wake an ops people at 2am just because a ping failed. I was referring to capturing the error as a field:

```rust
tracing::debug!(error = error as &dyn std::error::Error, "Ping failure");
```

For this, you'll also have to change `Some(error)` to `Some(ref error)`.